### ISSUE TITLE
"relay" crash

### ISSUE THREAD

**Issue by jcarnat** [OpenSMTPD/OpenSMTPD#35]
Trying to use the "relay as" feature, smtpd dies in openbsd 5.1

Setting 'accept for all relay as "@tumfatig.net"' in smtpd.conf leads to:
smtp: new client on listener: 0x3c00aba0
session_pickup: greeting client
scheduler_ramqueue: insert
888f8b99: from=ptijo@directory.tumfatig.net, size=421, nrcpts=1, proto=ESMTP, relay=1000@localhost [IPv6:::1]
smtp: 0x84e2a000: deleting session: done
scheduler_ramqueue: display
       scheduler_ramqueue: hosttree display
               host: [0x87cf2000] directory.tumfatig.net
                       batch: [0x8984c180] 00000000888f8b99
                               evpid: [0x8984c3c0] 888f8b99373d71cc
       scheduler_ramqueue: msgtree display
               msg: [0x8b859420] 00000000888f8b99
                       evp: [0x8984c3c0] 888f8b99373d71cc
       scheduler_ramqueue: queue display
               evpid: [0x8984c3c0] [batch: 0x8984c180], 888f8b99373d71cc
scheduler_ramqueue: next
scheduler_ramqueue: next: found
scheduler_ramqueue: next
scheduler_ramqueue: next: found
scheduler_ramqueue: next
scheduler_ramqueue: next: found
scheduler_ramqueue: remove
scheduler_ramqueue_remove: batch removed
scheduler_ramqueue: next
scheduler_ramqueue: next: nothing schedulable
runner: nothing to schedule, wake me up. zZzZzZ
mta: 0x8b856d00: new session for batch 4122582936
mta: 0x8b856d00: adding ptijo@directory.tumfatig.net from envelope 888f8b99373d71cc
dns: lookup mx "directory.tumfatig.net"
mta: 0x8b856d00: connecting to 192.168.0.67...
fatal: mta cannot create socket: Connection refused
lost child: mail transfer agent exited abnormally
lookup agent exiting
smtp server exiting
mail filter exiting
control process exiting
queue handler exiting
mail delivery agent exiting
runner handler exiting
parent terminating

Setting 'accept for all relay via mail.tumfatig.net as "@tumfatig.net"'  does not replace the recipient (nor does "user@domain.tld").

To sum up the discussion on ML: "So relay via doesn't crash but does nothing, and relay crashes."
